.github/workflows/main.yaml

```diff
@@ -3,26 +3,48 @@ on:
   push:
     branches:
       - "main"
-permissions:
-  contents: read
-  pull-requests: write
+
 env:
   ENVIRONMENT: 'development'
-  HUMCTL_VERSION: '0.23.2'
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+
       - uses: humanitec/setup-cli-action@v1
         with:
-          version: ${{ env.HUMCTL_VERSION }}
-      - name: Run Score
+          version: ${{ vars.HUMCTL_VERSION }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GH_TOKEN }}
+
+      - name: Build and push image
+        id: build_push
+        run: |
+          # Make sure repo name is all lowercase for docker push
+          GH_REPOSITORY=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
+
+          TAG=$GH_REPOSITORY:latest
+          IMAGE=ghcr.io/$TAG
+
+          # Set this label to connect the package to the GitHub repo to make it public
+          docker build . --tag $IMAGE  \
+            --label "org.opencontainers.image.source=https://github.com/$GH_REPOSITORY"
+          docker push $IMAGE
+          echo "IMAGE=$IMAGE" >> $GITHUB_OUTPUT
+
+      - name: Deploy Score
         run: |
           humctl score deploy \
```

Comment:
Let's add an explicit message/comment here too?

Reply:
Very good point. I added a message showing the latest git commit message. This works (I tested it) without the curl to the GH API, preventing a dependency to GitHub for any adopters.